SwiftRuntimeLibrary/SwiftMarshal/StructMarshal.cs

```diff
@@ -1546,6 +1565,17 @@ public IntPtr RetainNominalData (Type t, IntPtr p, int size)
 			return p;
 		}
 
+		public unsafe void RetainNominalData (SwiftMetatype metadata, byte [] p)
+		{
+			byte* tbuf = stackalloc byte [p.Length];
```

Comment:
Curious about why we need to explicitly allocate this on the stack?

Reply:
I believe this is an optimization, so we don't have to pressure the GC.

See: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/stackalloc

Reply:
Ah, makes sense. Otherwise it would be default allocated on the heap then gc'd. 😄

Reply:
Here's the long story: swift does NOT have a function call to say "take a pointer to this value type and bump any reference counts in it". It does have a call that says "take a pointer to this value type and copy it to this buffer, bumping the reference counts." So we do that into temporary memory since it goes away after the call. Stack allocation is super fast and the dispose is essentially free since it goes away when the stack frame gets deallocated. The Aristocrats!

Reply:
Super question, by the way.